File: leetcode_problems/typescript/src/lc2379_min_recolors.ts
function minimumRecolors(blocks: string, k: number): number {
  let n = blocks.length;
  let w = 0;
  let min = 0;
  for (let i = 0; i < k && i < n; i++) {
    if (blocks[i] === "W") {
      w++;
    }
  }
  min = w;
  for (let i = k; i < n; i++) {
    if (blocks[i - k] === "W") {
      w--;
    }
    if (blocks[i] === "W") {
      w++;
    }
    min = Math.min(min, w);
  }
  return min;
}

File: leetcode_problems/cpp/lc2379_min_recolors.cpp
#include <string>
#include <algorithm>
using std::string;

class Solution
{
public:
    int minimumRecolors(string blocks, int k)
    {
        int n = blocks.size();
        int w = 0;
        int min = 0;
        for (int i = 0; i < k && i < n; i++)
        {
            w += blocks[i] == 'W';
        }
        min = w;
        for (int i = k; i < n; i++)
        {
            w -= blocks[i - k] == 'W';
            w += blocks[i] == 'W';
            min = std::min(min, w);
        }
        return min;
    }
};